soteria-rust/lib/heap.ml

```diff
@@ -361,7 +361,8 @@ let load_global g ({ globals; _ } as st) =
 let borrow (ptr, meta) (ty : Charon.Types.ty)
     (kind : Charon.Expressions.borrow_kind) st =
   (* &UnsafeCell<T> are treated as raw pointers, and reuse parent's tag! *)
-  if Layout.is_unsafe_cell ty then Result.ok ((ptr, meta), st)
+  if (kind = BShared && Layout.is_unsafe_cell ty) || Tree_borrow.is_disabled ()
```

Comment:
Not sure how expensive Layout.is_unsafe_cell is, but `||` is short-circuiting in OCaml, so maybe the other way around would be quicker?

Reply:
Just looked, it's definitely more expensive in some cases to have `is_unsafe_cell` since you're recursively going through the structure